tests/test_requests.py

```diff
@@ -606,6 +626,27 @@ def test_unicode_header_name(self, httpbin):
     def test_pyopenssl_redirect(self, httpbin_secure, httpbin_ca_bundle):
         requests.get(httpbin_secure('status', '301'), verify=httpbin_ca_bundle)
 
+    def test_https_warnings(self, httpbin_secure, httpbin_ca_bundle):
+        """warnings are emitted with requests.get"""
+        if HAS_MODERN_SSL or HAS_PYOPENSSL:
+            warnings_expected = ('SubjectAltNameWarning', )
```

Comment:
It should happen once per connection. `httpbin_secure` does no caching, it literally returns a URL. Any caching would happen in `requests.get`, and `requests.get` explicitly does not connection pool, so this test should be fine.